### ISSUE TITLE
Consider disabling module auto-loading during IntelliSense

### ISSUE THREAD

**Issue by daviwil** [PowerShell/PowerShellEditorServices#438]
Module auto-loading during IntelliSense is causing some issues for people, including session hangs and IntelliSense slowness.  We use `Get-Command` to gather information about the currently active IntelliSense result.  This causes the module containing the command to get loaded, and if the module misbehaves somehow, it can damage the rest of the session.

We should consider setting the `$PSModuleAutoLoadingPreference` to `"none"` before running Get-Command so that IntelliSense details are only gathered for modules that the user has loaded.  To make sure we have cmdlet signature details, we could potentially just parse the tooltips of the completions if they're available without loading the module.

**Comment by tylerl0706** [comment]
Could this autoloading be causing https://github.com/PowerShell/vscode-powershell/issues/861 ?

**Comment by daviwil** [comment]
Nah, this is caused by the use of DSC.  For some reason, having `Import-DSCResource` in a script slows IntelliSense to a crawl.

**Comment by adamdriscoll** [comment]
@TylerLeonhardt - I know this is old but this is kind of a pain in the but for binary module developers. If I install UniversalDashboard, it will load it from the PSModulePath. If I specify a path in my script to load the module from a different directory, that won't work because even if I use -Force, the UD module has already been loaded from my path.

**Comment by rjmholt** [comment]
We run intellisense in the same runspace as the console so that `Register-ArgumentCompleter` and a few other things will work. We've looked into kicking intellisense into a different runspace/thread/process but PowerShell just doesn't support that today.

So turning off autoloading for intellisense while keeping it on in the console is slightly complicated. Essentially we need to do some reflection to reach in and [set autoloading preference variable](https://github.com/PowerShell/PowerShell/blob/42f21ae6a00feeebb0e6777d7d9c5062535d2860/src/System.Management.Automation/engine/CommandDiscovery.cs#L784-L785) here:

https://github.com/PowerShell/PowerShellEditorServices/blob/7ebc64e3a89849b7ffd6863e3296407bc6718095/src/PowerShellEditorServices/Services/Symbols/Vistors/AstOperations.cs#L35-L140

However, doing so might be a performance issue, since it means either executing PowerShell or doing some risky reflection in a multithreaded scenario.

**Comment by adamdriscoll** [comment]
Cool. Thanks for the info. I just wanted to bring up that this was an issue for me and at least one other user I know of.

**Comment by SeeminglyScience** [comment]
The other problem is how limited intellisense would be without autoloading.  By default, only `Microsoft.PowerShell.Core` and `PSReadLine` are imported, which only yields a total of 64 cmdlets.  You'd have to manually import a whole lot of modules to make intellisense useful.

One day I hope to see a future where PowerShell has a completion API that is entirely static, based on `MetadataReader` and AST analysis.  Until then, I usually just run `Restart PowerShell Session` before I manually import the WIP version of the module I'm working on.

**Comment by TylerLeonhardt** [comment]
Cc @SteveL-MSFT @daxian-dbw an interesting discussion here.

**Comment by TylerLeonhardt** [comment]
I also wanna chime in and say that the TEMP debugging feature which allows you to debug in a brand new integrated console (also which is fixed in the latest preview extension) is a perfect candidate for binary module debugging. All you have to do is use the:

`"createTemporaryIntegratedConsole": true` in a `launch` debug configuration.
